### PR TITLE
automate compilation and building of python package

### PR DESCRIPTION
**Pull request by bernstei**
Use [`scikit-build-core`](https://github.com/scikit-build/scikit-build-core) to automatically compile extension and build python package.  Refactor code taking advantage of that to make `Symmetrix` calculator more easily importable, an executable that does mace model conversions, and automatically do such conversions if a pytorch model is passed to the calculator.

### CODE REVIEW

File: symmetrix/python/symmetrix/cli/convert_mace.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+from ..convert_mace import extract_model_data 
+
+def main():
+    parser = ArgumentParser()
+    parser.add_argument("--species", "-Z", "-s", nargs="+", help="atomic numbers or chemical symbols to extract", default=[])
+    parser.add_argument("--head", "-H", help="Head to keep, ignored unless model is multihead. "
+                                             "Defaults to first non-PT head, same as mace.tools.script_utils.remove_pt_head")
+    parser.add_argument("--output_file", "-o", help="output filename")
+    parser.add_argument("model_file", help="torch model file")
```
[the remark this answers is not in the document]
Comment (alinelena) [reply symmetrix/python/symmetrix/cli/convert_mace.py:13]: practically with current design is one puts the model after --species list... will get an error.

[the remark this answers is not in the document]
Comment (alinelena) [reply symmetrix/python/symmetrix/cli/convert_mace.py:13]: that is true... but imagine the casual user... i prefer kiss design

[the remark this answers is not in the document]
Comment (bernstei) [reply symmetrix/python/symmetrix/cli/convert_mace.py:13]: The casual user will probably try `symmetrix_extract_mace MACE.model`. And actually, if MACE starts saving the elements that are supposed by a fine-tuned head, that'd be enough most of the time.

[the remark this answers is not in the document]
Comment (wcwitt) [reply symmetrix/python/symmetrix/cli/convert_mace.py]: I don't have a strong view (and I don't think @bernstei does either), so I am accepting the suggestion.
